sphinx/source/docs/user_guide/embed.rst

```diff
@@ -84,6 +84,12 @@ The returned ``<script>`` will look something like:
 
     </script>
 
+**Warning:** If you need to get the HTML components as strings in a notebook 
+setting, make sure all of your Bokeh-related code related to that plot is 
+kept in the same cell block. Otherwise, it is not recommended to use components 
+together with the `show` function, as prior user experience has shown that
+it can behave erratically.
```

Comment:
I would leave off the warning admonishment and just have a regular paragraph. I might also suggest simplifying the message (both here and above to):

> Note that in Jupyter Notebooks, it is not possible to use ``components`` and ``show`` in the same notebook cell.

Reply:
On it!